docs/jobs.md

```diff
@@ -0,0 +1,432 @@
+  },
+  "state": "running",      // queued, running, errored
+  "worker_id": "123456",   // or unknown if in the queue
+  "try_count": 1,          // number of time the job has been executed.
+                           // increased at the start of new execution
+  "queued_at": "2016-09-19T12:35:08Z",  // time of the queuing
+  "started_at": "2016-09-19T12:35:08Z", // time of first execution
+  "errors": [],            // list of errors
+  "output": {},            // output of the worker, if any
+}
+```
+
+Example and description of a job creation options — as you can see, the options are replicated in the `io.cozy.jobs` attributes:
+
+```js
+{
+  "priority": 3,         // priority from 1 to 100
+  "timeout": 60,         // timeout value in seconds
+  "retry_max_count": 3,  // maximum number of retry
+  "retry_delay": 10,     // retry delay in seconds between each try
+  "argument": {},        // arguments message
+}
+```
+
+
+Jobs API
+--------
+
+### GET /jobs/:job-id
+
+Get a job description given its ID.
+
+#### Status codes
+
+* 200 OK, when the job is found its description written
+* 404 Not Found, when the job has not been found (it either never existed or is finished)
+
+#### Request
+
+```http
+GET /jobs/123123 HTTP/1.1
+Accept: application/vnd.api+json
+```
+
+#### Response
+
+```json
+{
+  "data": {
+    "type": "io.cozy.jobs",
+    "id": "123123",
+    "rev": "1-12334",
+    "attributes": {
+      "worker": "sendmail",
+      "worker_id": "123123",
+      "launcher": "@cron",
+      "launcher_id": "4321",
+      "options": {
+        "priority": 3,
+        "timeout": 60,
+        "retry_max_count": 3,
+        "retry_delay": 10,
+        "argument": {},
+      },
+      "state": "running",
+      "worker_id": "123456",
+      "try_count": 1,
+      "queued_at": "2016-09-19T12:35:08Z",
+      "started_at": "2016-09-19T12:35:08Z",
+      "errors": [],
+      "output": {},
+    }
+  }
+}
+```
+
+
+### POST /jobs/queue/:worker-name
+
+Enqueue programmatically a new job.
+
+#### Request
+
+```http
+POST /jobs/queue/sendmail HTTP/1.1
+Accept: application/vnd.api+json
+
+{
+  "priority": 3,
+  "timeout": 60,
+  "retry_max_count": 3,
+  "retry_delay": 10,
+  "argument": {},
+}
+```
+
+#### Response
+
+```json
+{
+  "data": {
+    "type": "io.cozy.jobs",
+    "id": "123123",
+    "rev": "1-12334",
+    "attributes": {
+      "worker": "sendmail",
+      "worker_id": "123123",
+      "launcher": "@cron",
+      "launcher_id": "4321",
+      "options": {
+        "priority": 3,
+        "timeout": 60,
+        "retry_max_count": 3,
+        "retry_delay": 10,
+        "argument": {},
+      },
+      "state": "running",
+      "worker_id": "123456",
+      "try_count": 1,
+      "queued_at": "2016-09-19T12:35:08Z",
+      "started_at": "2016-09-19T12:35:08Z",
+      "errors": [],
+      "output": {},
+    },
+    "links": {
+      "self": "/jobs/123123"
+    }
+  }
+}
+```
+
+
+### GET /jobs/queue
+
+List the jobs in the queue.
```

Comment:
I would like the provide the list associated with metadata about its total size. How would you spec that with JSON-API... ?